cli/src/cli_util.rs

```diff
@@ -3701,6 +3701,23 @@ impl CliRunner {
             process_global_args_fn(ui, &matches)?;
         }
 
+        let subcommand_combined =
+            if let Some((subcommand, mut subcommand_matches)) = matches.subcommand() {
+                let mut subcommand_combined = String::from(subcommand);
+                while let Some((subcommand, new_matches)) = subcommand_matches.subcommand() {
+                    subcommand_combined.push(' ');
+                    subcommand_combined.push_str(subcommand);
+                    subcommand_matches = new_matches;
+                }
+                config = config_env
+                    .resolve_config_with_command(&raw_config, Some(&subcommand_combined))?;
+                migrate_config(&mut config)?;
+                ui.reset(&config)?;
+                Some(subcommand_combined)
+            } else {
+                None
```

Comment:
nit: perhaps, actual command name can be set to `""` if no subcommands exist. `None` means the command name is unknown, whereas the command is known to be unspecified here.